plugin/db/driver.go

```diff
@@ -492,3 +503,60 @@ func FormatParamNameInNumberedPosition(paramNames []string) string {
 	}
 	return fmt.Sprintf("WHERE %s ", strings.Join(parts, " AND "))
 }
+
+// IsAsterisksInTemplateValid checks if the pathTemplate is valid about asterisk.
+func IsAsterisksInTemplateValid(pathTemplate string) error {
+	const maxAsteriskNum = 2
+	if err := isMaxConsecutiveAsteriskValid(pathTemplate, maxAsteriskNum); err != nil {
+		return err
+	}
+	if err := isSingleAsteriskInTemplateValid(pathTemplate); err != nil {
+		return err
+	}
+	return isDoubleAsteriskInTemplateValid(pathTemplate)
+}
+
+// isSingleAsteriskInTemplateValid checks whether the single in file path template is valid.
+func isSingleAsteriskInTemplateValid(pathTemplate string) error {
+	return isMultipleTimesAsteriskInTemplateValid(pathTemplate, 1)
+}
+
+// isDoubleAsteriskInTemplateValid checks whether the consecutive double asterisks in file path template is valid.
+func isDoubleAsteriskInTemplateValid(pathTemplate string) error {
+	return isMultipleTimesAsteriskInTemplateValid(pathTemplate, 2)
+}
+
+// isMaxConsecutiveAsteriskValid returns true if the pathTemplate contains `n` consecutive asterisk at most.
+func isMaxConsecutiveAsteriskValid(pathTemplate string, n int) error {
+	re := regexp.MustCompile(strings.Repeat(`\*`, n+1))
+	if re.MatchString(pathTemplate) {
+		return errors.Errorf("path template %s contains more than %d asterisk", pathTemplate, n)
+	}
+	return nil
+}
+
+// isMultipleTimesAsteriskInTemplateValid checks whether the consecutive multiple asterisks in file path template is valid.
```

Comment:
nit: "consecutive" implies "multiple", same for other occurrences 
```suggestion
// isMultipleTimesAsteriskInTemplateValid checks whether the consecutive asterisks in file path template is valid.
```